File: Java/-09-.java
/**
 * Crea un programa se encargue de transformar un número
 * decimal a binario sin utilizar funciones propias del lenguaje que lo hagan directamente.
 **/
public class DecimalConverter {
    public Integer DtoB(Integer numero){
        Integer cociente = 0, residuo = 0;
        List<Integer> residuos = new ArrayList<>();
        do {
            cociente = Math.round(numero / 2);
            residuo = Math.round(numero % 2);
            residuos.add(residuo);

        } while (cociente != 0);
        residuos = residuos.reversed();
        StringBuilder sb = new StringBuilder();
        for (Integer n : residuos){
            sb.append(n);
        }
        return Integer.parseInt(sb.toString());
    }
}

File: C#/09.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _09
{
    /*
     * Crea un programa se encargue de transformar un número
     * decimal a binario sin utilizar funciones propias del lenguaje que lo hagan directamente.
     */
    internal class DecimalToBinary
    {
        public string convertir(int numero)
        {
            if (numero == 0) return "0";
            string binario = "";
            while (numero > 0)
            { 
                binario = (numero % 2) + binario;
                numero = numero / 2;
            }
            return binario;
        }
    }
}
